### ISSUE TITLE
[vLLM] gather_logprobs graph fails trace-insertion at opt_level=1

### ISSUE THREAD

**Issue by kmabeeTT** [tenstorrent/tt-xla#4387]
## Summary

vLLM's `gather_logprobs` graph fails compilation under metal trace (`enable_trace=True`) at `optimization_level=1`. Upcoming PR will hang the parent metal-trace feature and works around this one graph by (a) skipping the startup precompile and (b) routing sampled-token logprob post-processing through CPU when `enable_trace=True`. `optimization_level=0` compiles on-device correctly. The CPU fallback is covered by the regression test `tests/integrations/vllm_plugin/sampling/test_trace_logprobs.py::test_opt125m_trace_logprobs` at `opt_level=1`.

## Details

`gather_logprobs` uses `torch.gather(logits, -1, selected_token_ids)`, which lowers to `ttir.embedding`. With trace insertion enabled at `opt_level=1`, the pass fails:

```
error: 'ttnn.capture_or_execute_trace' op All output tensors of trace function must be on device.
%22 = "ttnn.from_device"(%arg1) : (tensor<1x1xsi32, ...dram>) ->
      tensor<1x1xsi32, ...system_memory> is not on device.
```

Same shape of failure as the RoPE case in #4220 — `from_device` is inserted on a runtime index tensor (here `selected_token_ids`, shape `1x1xsi32`), and the resulting host-side tensor becomes a trace output, violating the trace verifier.

## Repro

The test below passes today because of the CPU fallback in `model_runner.py`:

```bash
pytest -sv tests/integrations/vllm_plugin/sampling/test_trace_logprobs.py::test_opt125m_trace_logprobs
```

To reproduce the underlying compiler failure, remove the `enable_trace` branch that routes through CPU (the `TODO(#4387)` block at the `gather_logprobs` call site in `model_runner.py`) and rerun the test. The on-device path will then hit the trace-insertion error shown above.

## Desired fix

Either:
1. Trace-insertion pass handles `ttir.embedding` with an on-device index (generalizes the RoPE fix from #4220).
2. `torch.gather` with an on-device index lowers to something other than `ttir.embedding`.

## Follow-ups in tt-xla once fixed

- Remove the CPU fallback branch in `model_runner.py` (inline `TODO(#XXXX)` at the `gather_logprobs` call site).
- Remove the `enable_trace` guard around `_precompile_gather_logprobs` in `model_runner.py::capture_model`.
- Test at `opt_level=1` should keep passing via the on-device path — no test change needed.

==

(Will be opening tt-mlir issue with details soon to track debug + fix)
